tests/test_equivalence_checking.py

```diff
@@ -50,7 +50,30 @@ def test_try_parameter_combinations(tmp_path: Path) -> None:
 def test_find_counter_examples() -> None:
     """Test the function find_counter_examples."""
     num_bits = 6
-    num_counter_examples = 10
+    num_counter_examples = 3
+    res_string, counter_examples = equivalence_checking.create_condition_string(
+        num_bits=num_bits, num_counter_examples=num_counter_examples
+    )
+    shots = 512
+    delta = 0.7
+    found_counter_examples = equivalence_checking.find_counter_examples(
+        miter=res_string, num_bits=num_bits, shots=shots, delta=delta
+    )
+    if isinstance(found_counter_examples, list):
+        found_counter_examples.sort()
+    counter_examples.sort()
+    assert found_counter_examples == counter_examples
+
+    number_iterations = equivalence_checking.find_counter_examples(
+        miter=res_string, num_bits=num_bits, shots=shots, delta=delta, counter_examples=counter_examples
+    )
+
+    # 5 iterations were determined by multiple experiments in the case of 6 bits and 3 counter examples
+    assert number_iterations == 5
+
+    # Test, if the correct group of states gets identified as counter examples
+    num_bits = 6
+    num_counter_examples = 57
```

Comment:
Could you add a comment by such a high number is used here? I suspect it is to have >50% counter examples?